### ISSUE TITLE
Does Adam Optimization depend on batch size?

### ISSUE THREAD

**Issue by fancunwei95** [deepmind/ferminet#11]
Hello,

This is Cunwei and I am interested in ferminet and its applications. When we try to read and run the code, we have a minor problem about the minimize procedure in the code.  In qmc.py #264 the code looks like

```
optimize_step = functools.partial(
            optimizer.minimize,
            features,
            global_step=global_step,
            var_list=self.network.trainable_variables,
            grad_loss=grad_loss_clipped)

```

Thus, my question is that whether this gradient computation depends on the batch size for **Adam Optimizer**. I checked the tensorflow source code (not sure the version is what required here) and find the adam optimizer  tries to  dot product grad_loss and the features gradient. Thus, it seems that the gradients is extensive. But if  it is not, is this scale dealt with somewhere else? 

Thank you a lot for answering this question.

**Comment by dpfau** [comment]
The hyperparameters should not need to be changed with batch size. We use the mean of the local energies when computing the loss, so the gradient should not depend on batch size. Even so, methods like ADAM are self-tuning, and so should adapt to changes in batch size. If you are using the TensorFlow code, we would recommend using the KFAC optimizer rather than ADAM. If you only want to try ADAM, we would recommend using the JAX code (in the `jax` branch) as it is cleaner and easier to understand.

**Comment by fancunwei95** [comment]
Thank you for your suggestion. Since we are trying to test the code in pytorch and the converges in pytorch is not as quick as in the tensorflow code. 

After some investigation, we think the optimizer.minimize function only combines the features ($\log|\psi|$) and grad_loss ($E_L - \mathbb{E}(E_L)$) in a dot product way and the denominator of the batch size is not included and that is why we ask the question.

**Comment by dpfau** [comment]
Ah yes, I think there's a slight difference between the JAX and TF implementations. But because of the way both ADAM and KFAC normalize the updates, it shouldn't matter.
